shim/interfaces.go

```diff
@@ -80,6 +80,13 @@ type ChaincodeStubInterface interface {
 	// If the key does not exist in the state database, (nil, nil) is returned.
 	GetState(key string) ([]byte, error)
 
+	// GetMultipleStates returns the values of the specified `keys` from the
+	// ledger. Note that GetMultipleStates doesn't read data from the writeset, which
+	// has not been committed to the ledger. In other words, GetMultipleStates doesn't
+	// consider data modified by PutState that has not been committed.
+	// If the keys do not exist in the state database, (nil, nil) is returned.
```

Comment:
Is this actually the behaviour? If so... say I ask for keys `{ "a", "b", "c" }`, and "b" does not exist in the state database, which of the following is returned?

- A nil [][]byte.
- A slice of values for only keys "a" and "c" (2 values).
- A slice with values for key "a", nil and "c" (3 values with a nil []byte for the value of key "b").
- Something else...

I think I expected an error to be returned if any of the keys are not present in the state database.

Reply:
1. The order is important, the client then has to match the answers to the keys. Therefore, skips are not allowed.
2. When we request GetState and there is no key, it is not an error but nil that is returned. This is the default behaviour. If the chaincode returns an error when exchanging with a peer, it should immediately abort the computation and return an error to the user.

Reply:
And I'm just throwing in a challenge that already exists in ledger, I'm not the author of it. Its behaviour was not invented and implemented by me.

Reply:
I have checked what will be returned if b does not exist. 

without peer support - return [][]bite{[]byte{"val a"}, nil, []byte{"val c"}}

with peer support - return [][]bite{[]byte{"val a"}, []byte{}, []byte{"val c"}}

Reply:
It seems it is consistent with the GetState behaviour then. That sounds good to me. Just the doc needs to reflect the behaviour that it returns the values in the same order that the keys are provided, and any keys not present in the ledger return a nil value.

I think it is worth ensuring that the code does return a nil value for missing keys (like `GetStates` does), and not explicitly replace that nil with an empty []byte. Then perhaps the smart contract code does have some way of distinguishing between empty values and keys that are not present.

Reply:
Then there will be different behavior. if the peer does not support this operation, then nil is returned, and if it does, then an empty byte slice is returned. it turns out to be proto.Unmarshal. I brought it to a single format, but you can do the opposite, run through the response when the peer supports it and replace it with nil

Reply:
> I think it is worth ensuring that the code does return a nil value for missing keys (like `GetState` does), and not explicitly replace that nil with an empty []byte. Then perhaps the smart contract code does have some way of distinguishing between empty values and keys that are not present.

I agree. I'll redo it.

Reply:
> It seems it is consistent with the GetState behaviour then. That sounds good to me. Just the doc needs to reflect the behaviour that it returns the values in the same order that the keys are provided, and any keys not present in the ledger return a nil value.

done